src/Serializer/SerializerAbstract.php

```diff
@@ -38,22 +38,32 @@
     abstract public function meta(array $meta);
 
     /**
-     * Serialize the paginator.
+     * Serialize the cursor.
      *
-     * @param PaginatorInterface $paginator
+     * @param CursorInterface $cursor
      *
      * @return array
      */
-    abstract public function paginator(PaginatorInterface $paginator);
+    abstract public function cursor(CursorInterface $cursor);
 
     /**
-     * Serialize the cursor.
+     * Serialize the paginator.
      *
-     * @param CursorInterface $cursor
+     * @param PaginatorInterface $paginator
      *
      * @return array
      */
-    abstract public function cursor(CursorInterface $cursor);
+    public function paginator(PaginatorInterface $paginator)
+    {
+        $pagination = [
+            'pageNumber' => (int)$paginator->getCurrentPage(),
```

Comment:
This is a good fix, but that's not what I mean on the comment lol.

`'pageNumber' => (int)$paginator->getCurrentPage()`

instead of casting using `(int)`, should we update `PaginatorInterface` to guarantee that the functions always return ints?

`'pageNumber' => $paginator->getCurrentPage()`

Inside the interface

    public function getCurrentPage(): int